src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy

```diff
@@ -58,19 +59,37 @@ class JavaDistributionPlugin implements Plugin<Project> {
             }
         }
 
+        Task copyLauncherBinaries = project.tasks.create('copyLauncherBinaries', {
+            group = GROUP_NAME
+            description = "Creates go-java-launcher binaries."
+        }) << {
+            // TODO(rfink) Don't check these in, but fetch them from a published artifact?
+            ["javalauncher-linux-amd64", "javalauncher-darwin-amd64"].each { file ->
+                def dest = Paths.get("${project.buildDir}/scripts/${file}")
```

Comment:
I think the launch config options need to be editable (say to safely add env variables via deployment software). Users definitely need to be able to modify heap settings as well.

Fine putting it in var/launch or some other place, but don't think it can go in the service dir, since by mandate that's not editable.

Sent from my iPhone

On Jul 6, 2016, at 10:40 AM, Robert Fink <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/main/groovy/com/palantir/gradle/javadist/JavaDistributionPlugin.groovy<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_palantir_gradle-2Djava-2Ddistribution_pull_85-23discussion-5Fr69776704&d=DQMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=WP78GPbLtpuvYIwP3MU0cTu5Ks7YcCYBI_HSG6YaIEs&s=cGdCMvQ6AbO6HTPAPgynXiiWdsDzpN_Z1s3bkMiwn_c&e=>:

> @@ -58,19 +59,37 @@ class JavaDistributionPlugin implements Plugin<Project> {
>              }
>          }
>
> +        Task copyLauncherBinaries = project.tasks.create('copyLauncherBinaries', {
> +            group = GROUP_NAME
> +            description = "Creates go-java-launcher binaries."
> +        }) << {
> +            // TODO(rfink) Don't check these in, but fetch them from a published artifact?
> +            ["javalauncher-linux-amd64", "javalauncher-darwin-amd64"].each { file ->
> +                def dest = Paths.get("${project.buildDir}/scripts/${file}")


double-checking the reasoning here: the file is really part of the binary and is not intended to be edited by anyone directly. I think it's more correct to put it in bin/. as it stands, the stuff in the var/conf directory of the distribution is exactly the contents of the var/conf directory of the source, and conversely everything in bin/ is generated; I like that separation.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_palantir_gradle-2Djava-2Ddistribution_pull_85_files_c4d43f123e5aa09f0f0ef95acf96f33ccb0ec982-23r69776704&d=DQMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=WP78GPbLtpuvYIwP3MU0cTu5Ks7YcCYBI_HSG6YaIEs&s=tiIBWzyqjE8ir0YfbzNT6iaPiTDC9-9Ya5IEoa77hoY&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe_AAp3Q6rhZBB23sx6ldISZouRbx8Y-5FvSIks5qS-2DihgaJpZM4JFzUs&d=DQMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=WP78GPbLtpuvYIwP3MU0cTu5Ks7YcCYBI_HSG6YaIEs&s=UDAyzLjQMBRRRpIpHHJ33mkzPG54Wt8F2c3MsDiS5-w&e=>.